### ISSUE TITLE
False positive S2930 (C#8 using)

### ISSUE THREAD

**Issue by Hoffs** [SonarSource/sonar-dotnet#2744]
### Description

When using new C#8 syntax of `using var ....` to dispose `IDisposable` objects a warning is shown to dispose `FileStream` even though it is automatically disposed at the end. For some reason this warning is not shown for `MemoryStream`.

### Repro steps

```csharp
using System.IO;
namespace Test
{
    public class TestClass
    {
        private byte[] data;

        public TestClass(string filePath)
        {
            using var fs = new FileStream(filePath, FileMode.Open); // Dispose 'fs' when it is no longer needed.
            using var ms = new MemoryStream();
            fs.CopyTo(ms);
            data = ms.ToArray();
        }
    }
}
```

### Expected behavior

No warning.

### Actual behavior

Shows false positive warning.

### Known workarounds

Using old `using` syntax.

### Related information

* SonarAnalyzer.CSharp Version=7.17.0.9346 running as package analyzer in roslyn.

**Comment by costin-zaharia-sonarsource** [comment]
This was fixed in [8.3 release](https://github.com/SonarSource/sonar-dotnet/releases/tag/8.3.0.14607).

**Comment by lg2de** [comment]
Is there a PR related to this fix?

**Comment by costin-zaharia-sonarsource** [comment]
Hi @lg2de, 

> Is there a PR related to this fix?

Yes, you can find it here: https://github.com/SonarSource/sonar-dotnet/pull/3049
